File: programmers/2019 카카오 개발자 겨울 인턴쉽/튜플.js
function solution(s) {
  let answer = []
  // console.log(s)
  let replaced = s
    .split("")
    .map((str) => str.replace("{", "[").replace("}", "]"))
  let arr = JSON.parse(replaced.join("")).sort((a, b) => a.length - b.length)
  // console.log(arr)

  arr.forEach((a) => answer.push(...a.filter((item) => !answer.includes(item))))
  return answer
}

File: programmers/kakao/튜플.py


def solution(s):
    # {{, }}를 제거 후 },{ 으로 나누기
    data = s[2:-2].split("},{")
    # 길이 별로 오름차순 정렬
    data = sorted(data, key=lambda x: len(x))
    answer = []
    for item in data:
        # 각각의 원소로 분류 후
        print(item)
        item = list(map(int, item.split(",")))
        print(item)
        for value in item:
            # 포함되어 있지 않으면 input
            if value not in answer:
                answer.append(value)
    return answer